### ISSUE TITLE
StyleProvider: Allow defining a global nonce

### ISSUE THREAD

**Issue by jcheringer** [WordPress/gutenberg#60085]
## What problem does this address?

Some components ([Modal](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/modal/index.tsx#L261), [Popover.Slot](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/slot-fill/bubbles-virtually/fill.tsx#L56)) from the `@wordpress/components` package use their own [StyleProvider](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/style-provider/index.tsx), which, behind the scenes, creates an instance of the [CacheProvider](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/style-provider/index.tsx#L45) from the `@emotion/react` package.

However, there's no way to set a `nonce` for the CacheProvider, and in an application with more restrictive CSP (Content Security Policy) rules, the styles of [StyledComponents](https://emotion.sh/docs/styled) inside a Modal or a Popover are not loaded correctly.

![image](https://github.com/WordPress/gutenberg/assets/7811225/3194266f-60bd-4aa5-8889-dc832a8fab89)

I'm part of the team currently working on Gravatar.
We are facing this issue while trying to reutilize the [GradientPicker](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/gradient-picker/README.md) for a new feature.

## What is your proposed solution?

Expose a nonce property for the StyleProvider so an application using the WordPress components can define a nonce when necessary.

**Comment by mirka** [comment]
Makes sense. I think we can achieve this using our [context system](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#context-system):

1. Expose a `nonce` prop on `StyleProvider`.
2. Then connect `StyleProvider` to the context system, so a consumer can override the `nonce` prop on all `StyleProvider` instances wrapped within a `ContextSystemProvider`.

Would that solution work for your use case? Are any of you available to submit a PR?

**Comment by jcheringer** [comment]
Hey there!

I'm sorry I didn't respond to this at the appropriate time.

I'm bumping this issue again because we stumbled into it (again) while developing a Color Scheme feature for Gravatar.

The issue we are facing only happens during the development build. It works well in a production build.
Because of that, we ended up living with the CSP issue mentioned above in the past.


Dev build
![Image](https://github.com/user-attachments/assets/9b3cce22-f805-4d33-83b8-853d53a5b435)

Prod build
![Image](https://github.com/user-attachments/assets/fcc48dfc-10da-4399-98ca-9595eec9745c)

From a developer's perspective, it would be great if we could fix the issue.

> Would that solution work for your use case?

I believe that should work for us.

> Are any of you available to submit a PR?

Unfortunately, I don't have enough knowledge of the Gutenberg components' codebase to submit a PR.
I downloaded the code and played with it to see if I could get something to work, but I got stuck.
